src/api/hooks/useUpdateRegistrationStatus.ts

```diff
@@ -0,0 +1,38 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { setNotification } from '../../redux/notificationSlice';
+import { TerminatePublicationRequest, UnpublishPublicationRequest } from '../../types/registration.types';
+import { updateRegistrationStatus } from '../registrationApi';
+
+interface UpdateRequest {
+  registrationIdentifier: string;
+  data: UnpublishPublicationRequest | TerminatePublicationRequest;
+  onSuccess?: () => void;
+}
+
+export const useUpdateRegistrationStatus = () => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: ({ registrationIdentifier, data }: UpdateRequest) =>
+      updateRegistrationStatus(registrationIdentifier, data),
+    onSuccess: (response, variables) => {
+      dispatch(setNotification({ message: t('feedback.success.update_registration'), variant: 'success' }));
+      if (response.data) {
+        const key1 = ['registration', response.data.identifier, true];
+        if (queryClient.getQueryData(key1)) {
+          queryClient.setQueryData(key1, response.data);
+        }
+        const key2 = ['registration', response.data.identifier, false];
+        if (queryClient.getQueryData(key2)) {
+          queryClient.setQueryData(key2, response.data);
```

Comment:
Skjønner ikke helt hvorfor vi lager og overskriver key1 og 2 her. Er det de mest deskriptive navnene? Burde vi hatt en kommentar? Hvorfor sjekker vi om de er til stede før vi overskriver dem?

Reply:
VI gjør en PUT som returnerer oppdatert data. Så det vi gjør her er å sette responsen fra PUTen rett inn i query cache i stedet for å kjøre en ny GET som henter samme data som ligger i PUT.

Grunnen til at `true`/`false` er lagt til som siste element der er at `useFetchRegistration`-hooken har en query key `shouldNotRedirect` (et konsept jeg hater tbh, men endte sånn etter en tapt disksjon😅) hvor vi ved avpublisering av resultat sier at vi ikke skal redirecte, selv om APIet returnerer en redirect code. Dette er egentlig en lang og knotete historie jeg heller kan ta en show'n'tell på om det er interesse!

Men jeg sjekker om den finnes for å unngå at begge queryKeys får verdi, da det trolig bare er en av dem som er i bruk uansett